plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+            try {
+                detectedFractions = analyzer.detectNeededFractions()
+                        .stream()
+                        // filter out fractions already on classpath
+                        .filter(fd -> !allDependencies.contains(ArtifactSpec.fromFractionDescriptor(fd)))
+                        .collect(Collectors.toSet());
+                getLogger().info("Detected fractions:\n  {}",
+                        detectedFractions.stream().map(FractionDescriptor::gav)
+                                .collect(Collectors.joining(INDENTATION_DELIMITER)));
+            } catch (IOException e) {
+                throw new GradleException("Failed to scan for fractions.", e);
+            }
+
+            // exclude or add further fractions according to user config
+            extension.getFractions().forEach(f -> {
+                if (f.startsWith(EXCLUDE_PREFIX)) {
+                    FractionDescriptor toRemove = FractionDescriptor.fromGav(FractionList.get(), f.substring(1));
+                    getLogger().info("Excluding from detected fractions: {}", toRemove);
+                    detectedFractions.remove(toRemove);
+                } else {
+                    FractionDescriptor toAdd = FractionDescriptor.fromGav(FractionList.get(), f);
+                    getLogger().info("Adding to detected fractions: {}", toAdd);
+                    detectedFractions.add(toAdd);
+                }
+            });
+
+            // combine detected fractions with their dependent fractions
+            final Set<FractionDescriptor> fractionsWithDependencies = new HashSet<>(detectedFractions);
+            fractionsWithDependencies.addAll(detectedFractions.stream()
+                    .flatMap(f -> f.getDependencies().stream())
+                    .collect(Collectors.toSet()));
+
+            getLogger().info("Detected fractions including dependencies:\n  {}",
+                    fractionsWithDependencies.stream()
+                            .map(FractionDescriptor::gav)
+                            .sorted()
+                            .collect(Collectors.joining(INDENTATION_DELIMITER)));
+
+            final Set<ArtifactSpec> fractionsSpecs = fractionsWithDependencies.stream()
+                    .map(ArtifactSpec::fromFractionDescriptor)
+                    .collect(Collectors.toSet());
+
+            // resolve fraction artifacts and include into classpath entries
+            try {
+                List<Path> resolvedPaths = artifactResolvingHelper().resolveAll(fractionsSpecs).stream()
+                        .map(s -> s.file.toPath())
+                        .collect(Collectors.toList());
+                classpathEntries.addAll(resolvedPaths);
+            } catch (Exception e) {
+                throw new GradleException("Failed to resolve fraction dependencies.", e);
+            }
+        }
+
+        getLogger().info("Computed classpath entries:\n  {}",
```

Comment:
You might also want to consider reducing the number of logs going to `INFO`. I can imagine the logs getting super huge if we keep printing the dependencies / classpath entries for each step. You should consider printing the final computation alone at `INFO` and move the rest to `DEBUG`.